extension/src/providers/gotoProvider.ts

```diff
@@ -0,0 +1,67 @@
+import * as vscode from 'vscode';
+import { AutoLispExt } from '../extension';
+import { Sexpression } from '../format/sexpression';
+import { ReadonlyDocument } from '../project/readOnlyDocument';
+
+interface LocatedItem {
+	line: number;
+	column: number;
+	fspath: string;
+}
+
+export class AutolispDefinitionProvider implements vscode.DefinitionProvider{
+	async provideDefinition(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken): Promise<vscode.Location | vscode.Location[]> {
+		const editor: vscode.TextEditor = vscode.window.activeTextEditor;		
+		let selected: string =  editor.document.getText(editor.selection);
+		if (selected === "") {
+			ReadonlyDocument.getMemoryDocument(document).atomsForest.forEach(sexp => {
+				if (sexp instanceof Sexpression && sexp.contains(position)){
+					const atom = sexp.getAtomFromPos(position);
+					if (atom) {
+						selected = atom.symbol;
+					}		
+				}
+			});			
+		}
+		try {
+			return this.findGoToLocations(selected);
+		} catch (error) {
+			return;	// I don't believe this requires a localized error since VSCode has a default "no definition found" response
+		}
+	} 
+
+	// This was designed around the idea of "preference" for opened and workspace documents
+	private findGoToLocations(searchFor: string): vscode.Location[] {
+		const locations: vscode.Location[] = [];
```

Comment:
If the `searchFor` is built-in symbols like `mapcar`, it should return it early?

Reply:
Correct me if I am wrong, but all (or many) of those symbols could theoretically be redefined. If that is true, then the internally defined symbols don't necessarily mean anything from an _exclude_ perspective. If that is not true, then we should probably only exclude the `AutoLispExt.Data.internalLispFuncs` symbols.

Reply:
I am not sure this point. If the performance is not an issue, it is ok to do F12 on built-in symbols.

Reply:
Here is what I am talking about. AutoCAD doesn't explicitly protect those built in symbols, they could all theoretically have alternate definitions.
![image](https://user-images.githubusercontent.com/51800232/98550674-a32cfb80-2259-11eb-968e-8c801e62aa1f.png)